### PR TITLE
Linux/Input: fix evdev device list

### PR DESCRIPTION
**Pull request by Megamouse**
todo: fix order of events -> names

for #4116 and #5226

### CODE REVIEW

File: rpcs3/evdev_joystick_handler.cpp [hunk 1]
```diff
@@ -113,6 +113,23 @@ bool evdev_joystick_handler::Init()
 	return true;
 }
 
+std::string evdev_joystick_handler::get_device_name(const libevdev* dev)
+{
+	std::string name;
+	const auto uniqe = libevdev_get_uniq(dev);
+
+	if (uniqe != NULL)
```
Comment (TGEnigma) [inline rpcs3/evdev_joystick_handler.cpp]: use nullptr

File: rpcs3/evdev_joystick_handler.cpp [hunk 2]
```diff
@@ -113,6 +113,23 @@ bool evdev_joystick_handler::Init()
 	return true;
 }
 
+std::string evdev_joystick_handler::get_device_name(const libevdev* dev)
+{
+	std::string name;
+	const auto uniqe = libevdev_get_uniq(dev);
```
Comment (danilaml) [inline rpcs3/evdev_joystick_handler.cpp]: unique

File: rpcs3/evdev_joystick_handler.cpp [hunk 3]
```diff
@@ -632,7 +659,14 @@ int evdev_joystick_handler::add_device(const std::string& device, bool in_settin
 				close(fd);
 				continue;
 			}
-			const std::string name = et.name + ": " + libevdev_get_name(dev);
+
+			std::string name = get_device_name(dev);
+
+			if (unique_names.find(name) == unique_names.end())
```
Comment (randomstuff) [inline rpcs3/evdev_joystick_handler.cpp]: The name format is not the same for the first instance and the other ones AFAIU.

For consistency, I'd just use the same name format all the time:

    std::string name = get_device_name(dev);
    name = fmt::format("%d. %s", ++unique_names[name], name);

Comment (Megamouse) [reply rpcs3/evdev_joystick_handler.cpp]: That's possible. But what do others say.

Comment (Megamouse) [reply rpcs3/evdev_joystick_handler.cpp]: actually I think using numbers for all names is going to create inconsistencies. For example when you add different controllers on the next boot. You'd have to make sure the correct controllers always end up at the same numbers. This is not possible with evdev (I think). We get different events each Linux reboot.

Comment (randomstuff) [reply rpcs3/evdev_joystick_handler.cpp]: Well that's the whole thing: with `std::unordered_map<std::string, u32> unique_names;` the number are per device name. If you have two different devices they will be called "1. Ultimate Gamepad" and "1. Killer Gamepad" but if you have two same gamepads they will be called "1. Ultimate Gamepad" and "2. Ultimate Gamepad". Now the two game pads "Ultimate Gamepads" could be swapped between reboot but if they are really identical that's not a problem but it's actually a feature.

Comment (Megamouse) [reply rpcs3/evdev_joystick_handler.cpp]: you know... it's exactly what i do right now.
only i don't use 1.
did you even try?

Comment (randomstuff) [reply rpcs3/evdev_joystick_handler.cpp]: OK, I failed to parse you comment because I assumed you was not the author of the PR :)

**Comment by Megamouse** [comment]
closed because of upcoming fixes. I think it makes no sense to merge this before.

**Pull request merged**